packages/keycloak-user-management/src/components/CreateEditUserGroup/utils.tsx

```diff
@@ -0,0 +1,184 @@
+import { KeycloakService } from '@opensrp/keycloak-service';
+import { store } from '@opensrp/store';
+import { sendErrorNotification, sendSuccessNotification } from '@opensrp/notifications';
+import { ERROR_OCCURED } from '../../lang';
+import {
+  KEYCLOAK_URL_ASSIGNED_ROLES,
+  KEYCLOAK_URL_AVAILABLE_ROLES,
+  KEYCLOAK_URL_USER_GROUPS,
+} from '../../constants';
+import { KeycloakUserRole } from '../../ducks/userRoles';
+import { fetchKeycloakUserGroups, KeycloakUserGroup } from '../../ducks/userGroups';
+
+/**
+ * Fetch available roles
+ *
+ * @param {string} groupId - user group id
+ * @param {string} keycloakBaseURL - keycloak API base URL
+ * @param {Function} setRolesAction method to set state for selected actions
+ */
+export const fetchAvailableRoles = async (
+  groupId: string,
+  keycloakBaseURL: string,
+  setRolesAction: (role: KeycloakUserRole[]) => void
+) => {
+  const keycloakService = new KeycloakService(
+    `${KEYCLOAK_URL_USER_GROUPS}/${groupId}${KEYCLOAK_URL_AVAILABLE_ROLES}`,
+    keycloakBaseURL
+  );
+
+  return await keycloakService
+    .list()
+    .then((response: KeycloakUserRole[]) => {
+      setRolesAction(response);
+    })
+    .catch((_: Error) => {
+      sendErrorNotification(ERROR_OCCURED);
+    });
+};
+
+/**
+ * Fetch assigned roles
+ *
+ * @param {string} groupId - user group id
+ * @param {string} keycloakBaseURL - keycloak API base URL
+ * @param {Function} setRolesAction method to set state for selected actions
+ */
+export const fetchAssignedRoles = async (
+  groupId: string,
+  keycloakBaseURL: string,
+  setRolesAction: (role: KeycloakUserRole[]) => void
+) => {
+  const keycloakService = new KeycloakService(
+    `${KEYCLOAK_URL_USER_GROUPS}/${groupId}${KEYCLOAK_URL_ASSIGNED_ROLES}`,
+    keycloakBaseURL
+  );
+
+  return await keycloakService
+    .list()
+    .then((response: KeycloakUserRole[]) => {
+      setRolesAction(response);
+    })
+    .catch((_: Error) => {
+      sendErrorNotification(ERROR_OCCURED);
+    });
+};
+
+/**
+ * Fetch effective roles
+ *
+ * @param {string} groupId - user group id
+ * @param {string} keycloakBaseURL - keycloak API base URL
+ * @param {Function} setRolesAction method to set state for selected actions
+ */
+export const fetchEffectiveRoles = async (
```

Comment:
`keycloakService`, `fetchAvailableRoles` and this method are very similar, seems like there is room to dry it more.